### ISSUE TITLE
Make StatusButton more generic

### ISSUE THREAD

**Issue by harbaum** [LLK/scratch-vm#2130]
Setting "showStatusButton" in an extension makes the little status button dot appear right of the extensions name in the toolbar:

![image](https://user-images.githubusercontent.com/9105084/56473594-bdcc3380-646d-11e9-92fa-718fbaff1f79.png)

This is tightly coupled to scratchlink and enabling this activates major parts of the scratchlink.

There are various other ways to interact with hardware than scratchlink. WebUSB and Web Bluetooth are just two of them. I am working on a WebUSB based setup and would like to be able to control this status button and use it for hardware setup related things. The current setup does not allow this and requires me to mess with the scratch dom directly to setup some custom button to display the hardware status and e.g. connect and disconnect it.

Furthermore for security reasons many frameworks like WebUSB require the hardware setup to be triggered by a manual user interaction (usually called a user gesture in this context). So it would be great if any callback provided by this button would still be flagged as "user gesture" so the button can be used to setup e.g. a WebUSB device.

**Comment by apple502j** [comment]
No, WebUSB is not popular and not all browsers support it. You should not make your extensions Chromium-only.

**Comment by harbaum** [comment]
I always enjoy people telling me what to do and what not to do. WebUSB is at least supported by chrome and most derived browsers on windows, linux, macos and android. Scratchlink for example does not run on any device i own. I cannot see how that is better.

This request is in no way limited to any technology. Instead it opens the door to whatever technology you like even other scratchlink like websocket based proxies.

**Comment by towerofnix** [comment]
(All things considered, [WebUSB](https://wicg.github.io/webusb/) is *not* a proper standard yet - it's still a draft. Yes, browsers (aka chromium) have begun to implement it, but that doesn't mean it's necessarily stable and certainly doesn't mean it's ready for general use. But that's not totally related to the discussion of this feature request.)

**Comment by harbaum** [comment]
I fully agree that WebUSB is by no means perfect. But for me it's better than Scratchlink which does not work at all for me.

And to be honest: Being able to control usb hardware from within scratch running in the browser of a smartphone _is_ impressive.

But I am not asking for WebUSB support in Scratch. I am asking for generic support of arbitrary hardware interfaces.
